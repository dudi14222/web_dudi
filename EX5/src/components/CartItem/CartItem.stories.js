import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import {
  withKnobs,
  text, boolean, number, object,
} from '@kadira/storybook-addon-knobs';
import CartItem from './';


const storiesCartItem = storiesOf('CartItem', module);
storiesCartItem.addDecorator(withKnobs);

storiesCartItem.addWithInfo('with some props', '', () => {
  const story =
    <div className="container">
      <div className="table-responsive">
        <table className="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Price</th>
              <th>Qty</th>
              <th></th>
              <th></th>
            </tr>
          </thead>
          <tbody>

            <CartItem
              name={text('name', 'name1')}
              id={number('item id', 1)}
              price={text('price', '$20')}
              quantity={number('quantity', 6)}
              removeItem={action('clicked')}
              imageUrl={text('image url', '/images/passo-tonale.jpg')}
            />           

          </tbody>
        </table>
      </div>
    </div>

  return story;
});











