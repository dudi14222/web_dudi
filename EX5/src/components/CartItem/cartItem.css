@media (max-width:669px) {
    .btn-remove-item{
        display: none;
    }
}
@media (min-width:670px) {
    .lbl-remove-item{
        display: none;
    }
}
.cart-item-image{
    width: 40px;
    height: 40px;
}
