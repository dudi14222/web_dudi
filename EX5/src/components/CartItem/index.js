import React from 'react';

const CartItem = ({name, id, price, quantity, removeItem}) => (
    <tr>
        <td>{name}</td>
        <td>{price}</td>
        <td>{quantity}</td>
        <td><button type="button" className="btn btn-primary" onClick={() => removeItem(id)}>Remove</button></td>                
    </tr>    
);
export default CartItem;
