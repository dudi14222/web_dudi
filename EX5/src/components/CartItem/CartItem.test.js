import CartItem from '../CartItem';
import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import { shallow } from "enzyme";



describe('CartItem snapshot tests', () => {
    it('test snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <CartItem />
            </MemoryRouter>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })
})

describe('CartItem shallow tests', () => {
    it('should call remove item when clicking on remove', () => {
        let counter = 0;
        const prop = {name: 'n', id: 5, price: '1$', quantity: '3', imageUrl: ''
        ,removeItem: (id)=>(counter = counter + id)
        }
        const wrapper = shallow((
            <CartItem {...prop}/>
        )); 

        const button = wrapper.find(".td-remove");
        button.simulate('click');   
        expect(counter).toBe(5); 
    });
})



