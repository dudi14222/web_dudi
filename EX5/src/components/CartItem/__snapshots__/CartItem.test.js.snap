// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem snapshot tests test snapshot 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td>
    <img
      className="cart-item-image"
      src={undefined}
    />
     
  </td>
  <td
    className="td-remove"
    onClick={[Function]}
  >
    <label
      className="lbl-remove-item glyphicon glyphicon-remove"
    />
    <button
      className="btn btn-primary btn-remove-item"
      type="button"
    >
      Remove
    </button>
  </td>
</tr>
`;
